plugins/package-managers/gradle-inspector/src/main/kotlin/GradleDependencyHandler.kt

```diff
@@ -83,17 +83,17 @@ internal class GradleDependencyHandler : DependencyHandler<OrtDependency> {
             }
         }
 
-        val isMetadataOnly = dependency.extension == "pom" || isSpringMetadataProject
+        val hasNoArtifacts = dependency.pomFile == null || isSpringMetadataProject
```

Comment:
The `https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-core/1.5.2/kotlinx-coroutines-core-1.5.2.jar` IIUC only contains `kotlinmetadata` files. Do you know how these may get consumed?

Reply:
These get consumed by the `kotlin-multiplaform` Gradle plugin. Still, that artifact is not "metadata-only" in the ORT sense, as it comes with a [sources artifact](https://repo1.maven.org/maven2/org/jetbrains/kotlinx/kotlinx-coroutines-core/1.5.2/kotlinx-coroutines-core-1.5.2-sources.jar) that actually contains `.kt` files that can be scanned.

Reply:
Should the commit message be adjusted to no more say as rationale "still come with associated artifacts" ?
...because the existance of an artifact does not imply `metadata_only==false`, but what counts is the content.

Here, the rationale seems to even refer to the contents of the sources jar.

Reply:
> because the existance of an artifact does not imply metadata_only==false, but what counts is the content.

Actually, that does not match my understanding of `isMetadataOnly`. When you introduced the property (still under the name `isMetaDataOnly`) in e6870e7484d7ee7b668b347dc081763f5e7a69ea, the commit message said

    To indicate whether the package is just meta data and there is no
    corresponding source code to be examined. That property is supposed to
    be used by the downloader, scanner and the evaluator rules to determine
    whether the download, scan or the evaluation should be skipped.

So we should have `isMetadataOnly == true` if there is no corresponding source code, which in turn means to me that if there is corresponding source code, we should have `isMetadataOnly == false` in order to not skip the scanning.

I totally see that where could be "metadata-only" artifacts that have more metadata than just a POM, and that additional metadata comes in artifacts / archives that are not worth scanning. But covering that case was to my understanding not the original purpose of that property. That is also one of the reasons why I'm still a bit reluctant to introduce your proposed [`noSourcesAvailable` property](https://github.com/oss-review-toolkit/ort/issues/7808). Note that I'm not again changing / adjusting semantics for those properties here, but if we do so, we should first ensure that the code currently treats `isMetadataOnly` in the same way as it should be treated after the introduction of `noSourcesAvailable`.

But I'd like to avoid the discussion about what the semantics of `isMetadataOnly` *should* be as much as possible as part of this PR. *Currently*, it's just a flag to skip scanning as there is no source code to scan (which, again, overlap with the planned `noSourcesAvailable` semantics).